SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialActorUtils.h

```diff
@@ -2,18 +2,43 @@
 
 #pragma once
 
-#include "CoreMinimal.h"
+#include "EngineClasses/SpatialNetConnection.h"
 
 #include "Components/SceneComponent.h"
+#include "Containers/Array.h"
+#include "Containers/UnrealString.h"
 #include "Engine/EngineTypes.h"
-#include "EngineClasses/SpatialNetConnection.h"
 #include "GameFramework/Actor.h"
 #include "GameFramework/Controller.h"
 #include "GameFramework/PlayerController.h"
+#include "Math/Vector.h"
 
 namespace SpatialGDK
 {
 
+// We need the optional IgnoredActor argument because Actor deletions are broadcast before
+// SetOwner is called and ownership hierarchies are updated.
+inline TArray<AActor*> GetOwnershipHierarchyPath(const AActor* Actor, const AActor* IgnoredActor = nullptr)
+{
+	check(Actor != nullptr);
+
+	AActor* OwnerIterator = Actor->GetOwner();
+	if (OwnerIterator == nullptr)
+	{
+		return TArray<AActor*>();
```

Comment:
This might make it less likely to use RVO. If you are always allocating the array I'd define this at the top and always return it. Just happening to put nothing in it if there is nothing to put in it.

Reply:
(Generally avoid branching returns if you want to elide stuff)